extra_geom/detectors.py

```diff
@@ -1682,19 +1682,24 @@ def from_origin(cls, origin=(0, 0), asic_gap=None, unit=None):
         if unit is None:
             unit = cls.pixel_size
         if asic_gap is None:
-            asic_gap = cls.asic_gap
+            asic_gap_h = asic_gap_v = cls.asic_gap
+        elif isinstance(asic_gap, Collection):
+            asic_gap_h = asic_gap[0]
+            asic_gap_v = asic_gap[1]
```

Comment:
```suggestion
            asic_gap_h, asic_gap_v = asic_gap
```

This gives us free errors if you pass a longer sequence.